ros2doctor/ros2doctor/verb/call.py

```diff
@@ -0,0 +1,260 @@
+
+    This command can be invoked on multiple hosts to confirm that they can talk to each other
+    by using talker/listener, multicast send/receive to check topic discovering and
+    UDP communication.
+    This command outputs a summary table of msgs statistics at a custom rate(Hz).
+    """
+
+    def add_arguments(self, parser, cli_name):
+        parser.add_argument(
+            'topic_name', nargs='?', default='/canyouhearme',
+            help="Name of ROS topic to publish to (default: '/canyouhearme')")
+        parser.add_argument(
+            'time_period', nargs='?', default=0.1,
+            help='Time period to publish/send one message (default: 0.1s)')
+        parser.add_argument(
+            'duration', nargs='?', default=20, type=positive_int,
+            help='How long this process runs (default: 20s)')
+        parser.add_argument(
+            '-r', '--rate', metavar='N', type=float, default=1.0,
+            help='Rate in Hz to print summary table (default: 1.0)')
+        parser.add_argument(
+            '--ttl', type=positive_int,
+            help='TTL for multicast send (default: None)')
+
+    def main(self, *, args):
+        global summary_table
+        summary_table = SummaryTable()
+        rclpy.init()
+        executor = SingleThreadedExecutor()
+        pub_node = Talker(args.topic_name, args.time_period)
+        sub_node = Listener(args.topic_name)
+        executor.add_node(pub_node)
+        executor.add_node(sub_node)
+        try:
+            prev_time = time.time()
+            timeout = time.time() + args.duration
+            # pub/sub thread
+            exec_thread = threading.Thread(target=executor.spin)
+            exec_thread.start()
+            while time.time() < timeout:
+                # print table at user determined rate
+                if (time.time() - prev_time > float(1/args.rate)):
+                    summary_table.format_print_summary(args.topic_name, args.rate)
+                    summary_table.reset()
+                    prev_time = time.time()
+                # multicast threads
+                send_thread = threading.Thread(target=_send, kwargs={'ttl': args.ttl})
+                send_thread.daemon = True
+                receive_thread = threading.Thread(target=_receive)
+                receive_thread.daemon = True
+                receive_thread.start()
+                send_thread.start()
+                time.sleep(args.time_period)
+        except KeyboardInterrupt:
+            pass
+        finally:
+            executor.shutdown()
+            rclpy.shutdown()
+            pub_node.destroy_node()
+            sub_node.destroy_node()
+
+
+class Talker(Node):
+    """Initialize talker node."""
+
+    def __init__(self, topic, time_period, *, qos=10):
+        super().__init__('ros2doctor_talker')
+        self.i = 0
+        self.pub = self.create_publisher(String, topic, qos)
+        self.timer = self.create_timer(time_period, self.timer_callback)
```

Comment:
nitpick: I think all three of these variables can be prefixed with an underscore since they are implementation details.